cmd/find.go

```diff
@@ -385,126 +381,95 @@ func watchEvents(ctx *cli.Context, clnt Client, params watchParams, alias string
+		i, e = strconv.Atoi(duration[:len(duration)-1])
+		if e != nil {
+			// if we still cant parse, user input is invalid, return error.
+			return time.Time{}, probe.NewError(e)
+		}
+		i = i * conversion[strings.ToLower(p)]
 	}
 
-	// if cant parse check for conversion multiplier
-	t = pattern[len(pattern)-1:]
-	i, err = strconv.Atoi(pattern[:len(pattern)-1])
+	now := UTCNow()
 
-	// if still cant parse, user input is invalid, error will be
-	// thrown by method which called TimeHelper
-	if err != nil {
-		return 0, err
-	}
+	// Find all time in which the time in which the object was just created is after the current time
+	t := time.Date(now.Year(), now.Month(), now.Day()-i, now.Hour(), now.Minute(), 0, 0, time.UTC)
 
 	// if we reach this line, user has passed a valid alphanumeric string
-	return i * conversion[strings.ToLower(t)], nil
+	return t, nil
 }
 
-// GetPurl is used in conjunction with the {url} substitution argument to return presigned URLs
-func GetPurl(path string) string {
+// getShareURL is used in conjunction with the {url} substitution
+// argument to generate and return presigned URLs, returns error if any.
+func getShareURL(path string) string {
 	targetAlias, targetURLFull, _, err := expandAlias(path)
-	fatalIf(err, "Error with expand alias")
-	clnt, err := newClientFromAlias(targetAlias, targetURLFull)
-	fatalIf(err, "Error with newClientFromAlias")
-
-	isIncomplete := false
-
-	objectsCh := make(chan *clientContent)
-
-	content, err := clnt.Stat(isIncomplete)
-
-	fatalIf(err, "Error with client stat")
+	fatalIf(err.Trace(path), "Unable to expand alias")
 
-	// piping all content into the object channel to be processed
-	go func() {
-		defer close(objectsCh)
-		objectsCh <- content
-	}()
+	clnt, err := newClientFromAlias(targetAlias, targetURLFull)
+	fatalIf(err.Trace(targetAlias, targetURLFull), "Unable to newClientFromAlias")
 
-	// get content from channel to be converted into presigned URL
-	for content := range objectsCh {
-		fatalIf(content.Err, "Error with content")
+	content, err := clnt.Stat(false)
+	fatalIf(err.Trace(targetURLFull, targetAlias), "Unable to lookup file/object.")
 
-		if content.Type.IsDir() {
-			continue
-		}
+	// Skip if its a directory.
+	if content.Type.IsDir() {
+		return ""
+	}
 
-		objectURL := content.URL.String()
-		newClnt, err := newClientFromAlias(targetAlias, objectURL)
-		fatalIf(err, "Error with newClientFromAlias")
+	objectURL := content.URL.String()
+	newClnt, err := newClientFromAlias(targetAlias, objectURL)
+	fatalIf(err.Trace(targetAlias, objectURL), "Unable to initialize new client from alias.")
 
-		// set default expiry for each url (point of no longer valid), to be 7 days
-		expiry := time.Duration(604800) * time.Second
-		shareURL, err := newClnt.ShareDownload(expiry)
-		fatalIf(err, "Error with ShareDownloa")
+	// Set default expiry for each url (point of no longer valid), to be 7 days
+	shareURL, err := newClnt.ShareDownload(time.Duration(604800) * time.Second)
```

Comment:
This can be
shareURL, err := newClnt.ShareDownload(humanize.Week.Seconds())

Reply:
Actually we can make this a constant.